File: 백준/Bronze/14626. ISBN/ISBN.js
const input = require('fs').readFileSync(0).toString().trim();
const isbn = input.split('');

let totalSum = 0;
let targetWeight = 0;

for (let i = 0; i < 13; i++) {
    let weight = (i % 2 === 0) ? 1 : 3;
    
    if (isbn[i] === '*') {
        targetWeight = weight;
    } else {
        totalSum += Number(isbn[i]) * weight;
    }
}

for (let x = 0; x < 10; x++) {
    if ((totalSum + (targetWeight * x)) % 10 === 0) {
        console.log(x);
        break;
    }
}

File: 백준/Bronze/14626. ISBN/ISBN.py
import sys
input = sys.stdin.readline

isbn = input().strip()

total_sum = 0
target_weight = 0

for i in range(13):
    weight = 1 if i % 2 == 0 else 3
    
    if isbn[i] == '*':
        target_weight = weight
    else:
        total_sum += int(isbn[i]) * weight

for x in range(10):
    if (total_sum + target_weight * x) % 10 == 0:
        print(x)
        break